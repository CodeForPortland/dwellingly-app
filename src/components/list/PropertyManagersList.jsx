import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

function PropertyManagersList({ url, items }) {
  return (
    <div className="list-group">
      {items &&
        items.length > 0 &&
        items.map(item => {
          const { id, name, tickets, tenants } = item;
          return (
            <Link
              key={id}
              to={`${url}/${id}`}
              className="list-group__item list-group__item--basic">
              <div className="contact-group contact-group--basic">
                <h3 className="title">{name}</h3>
                <p>
                  {`${tenants.length} ${
                    tenants.length === 1 ? 'Tenant' : 'Tenants'
                  }`}{' '}
                  |{' '}
                  {`${tickets.length} ${
                    tickets.length === 1 ? 'Ticket' : 'Tickets'
                  }`}
                </p>
              </div>
            </Link>
          );
        })}
    </div>
  );
}

PropertyManagersList.propTypes = {
  items: PropTypes.arrayOf(PropTypes.shape({})),
  url: PropTypes.string.isRequired
};

PropertyManagersList.defaultProps = {
  items: []
};

export default PropertyManagersList;
