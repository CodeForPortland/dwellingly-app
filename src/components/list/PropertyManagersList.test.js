import React from 'react';
import renderer from 'react-test-renderer';
import PropertyManagersList from './PropertyManagersList';

describe('PropertyManagersList Component', () => {
  it('renders empty div if no items exist', () => {
    const PropertyManagersListComponent = renderer.create(<PropertyManagersList />).toJSON();
    expect(PropertyManagersListComponent).toMatchSnapshot();
  });
});
