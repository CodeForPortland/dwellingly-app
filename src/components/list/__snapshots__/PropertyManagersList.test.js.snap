// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyManagersList Component renders empty div if no items exist 1`] = `
<div
  className="list-group"
/>
`;
